pkg/stub/handler.go

```diff
@@ -162,3 +214,98 @@ func getDaemonSetOwnerRef(ds *appsv1.DaemonSet) (metav1.OwnerReference, error) {
 	}
 	return or, nil
 }
+
+// syncServiceUpdate synchronizes the status of the ClusterIngress that
+// corresponds to the service with the service's status.
+func (h *Handler) syncServiceUpdate(service *corev1.Service, deleted bool) (ret error) {
+	if len(service.Name) < 8 || service.Name[0:7] != "router-" {
+		return nil
+	}
+
+	ciName := service.Name[7:]
+
+	ciNamespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		logrus.Fatalf("Failed to get watch namespace: %v", err)
+	}
+
+	ci := &ingressv1alpha1.ClusterIngress{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterIngress",
+			APIVersion: "ingress.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ciName,
+			Namespace: ciNamespace,
+		},
+		// TODO: Add a label to the ClusterIngress and use a label
+		// selector here.
+	}
+	if err := sdk.Get(ci); err != nil {
+		return fmt.Errorf(
+			"could not get ClusterIngress for service %q: %v",
+			service.Name, err)
+	}
+
+	oldCi := ci
+	ci = ci.DeepCopy()
+	defer func() {
+		err := ci.UpdateStatus(oldCi)
+		ret = kerrors.NewAggregate([]error{ret, err})
+	}()
+
+	if deleted {
+		ci.SetStatusAvailableCondition(false, "service was deleted")
+
+		return nil
+	}
+
+	ep := &corev1.Endpoints{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Endpoints",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      service.Name,
+			Namespace: service.Namespace,
+		},
+	}
+	if err := sdk.Get(ep); err != nil {
+		return fmt.Errorf("could not get endpoints for service %q: %v",
+		                  service.Name, err)
+	}
+
+	numSubsets := len(ep.Subsets)
+	if numSubsets == 0 {
+		ci.SetStatusAvailableCondition(false,
+			"service has no endpoints")
+
+		return nil
+	}
+
+	serviceReady := false
+	for i := 0; i < numSubsets; i++ {
+		if len(ep.Subsets[i].Addresses) > 0 {
+			serviceReady = true
+			break
+		}
+	}
+	if !serviceReady {
+		ci.SetStatusAvailableCondition(false,
+		                               "service endpoints not ready")
+
+		return nil
+	}
+
+	if len(service.Status.LoadBalancer.Ingress) == 0 {
+		ci.SetStatusAvailableCondition(false,
+		                               "service has no ingresses")
+
+		return nil
+	}
+
+	ci.SetStatusAvailableCondition(true, "")
+	ci.Status.LoadBalancer.Ingress = service.Status.LoadBalancer.Ingress
```

Comment:
Set this field only when ci.Spec.HighAvailability.Type == ingressv1alpha1.CloudClusterIngressHA ?